### PR TITLE
Make counters in coroutines optional

### PR DESCRIPTION
**Pull request by msimberg**
Makes the counters optional and turns them off by default. Saves us some cache misses when creating and rebinding threads. It makes a small but noticeable difference in the `future_overheads` benchmark.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -602,6 +602,13 @@ if(HPX_WITH_THREAD_STEALING_COUNTS)
   hpx_add_config_define(HPX_HAVE_THREAD_STEALING_COUNTS)
 endif()
 
+hpx_option(HPX_WITH_COROUTINE_COUNTS BOOL
```
Comment (hkaiser) [inline CMakeLists.txt]: nitpick: should we name this `..._COUNTERS` instead?

Comment (biddisco) [reply CMakeLists.txt]: yes

Comment (msimberg) [reply CMakeLists.txt]: Counters was my first choice but there was prior art with counts... There are actually examples of both. `HPX_WITH_IO_COUNTERS`, `HPX_WITH_THREAD_STEALING_COUNTS`. I'm absolutely okay with changing it to counters.

Comment (msimberg) [reply CMakeLists.txt]: I changed the name to `HPX_WITH_COROUTINE_COUNTERS`.

**Pull request merged**